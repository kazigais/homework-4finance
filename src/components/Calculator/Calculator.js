import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
  Header, Button, Input, Label, Segment, Message,
} from 'semantic-ui-react';
import { FaDollarSign } from 'react-icons/fa';
import styles from './Calculator.module.scss';
import CalendarInput from '../CalendarInput/CalendarInput';

class Calculator extends PureComponent {
  static propTypes = {
    onSubmit: PropTypes.func,
    title: PropTypes.string,
    errorMessage: PropTypes.string,
    date: PropTypes.instanceOf(Date),
    amount: PropTypes.number,
  }

  static defaultProps = {
    onSubmit: () => {},
    title: '',
    errorMessage: undefined,
    date: new Date(),
    amount: undefined,
  }

  constructor(props) {
    super(props);
    const { amount, date } = props;
    this.state = {
      amount,
      date,
    };
  }

  setDate = (date) => {
    this.setState({ date });
  }

  setAmount = (amount) => {
    this.setState({ amount });
  }

  render() {
    const { onSubmit, title, errorMessage } = this.props;

    return (
      <Segment className={styles['form-segment']} raised>
        <Header as="h1">{title}</Header>
        {errorMessage && (
          <Message negative>
            <Message.Header>{errorMessage}</Message.Header>
          </Message>
        )}
        <Input
          className={styles.input}
          onChange={(e) => this.setAmount(e.target.value)}
          labelPosition="right"
          type="text"
          placeholder="Amount"
        >
          <Label basic><FaDollarSign /></Label>
          <input />
          <Label>.00</Label>
        </Input>
        <CalendarInput
          value={this.state.date}
          onChange={(value) => this.setDate(value)}
        />
        <Button
          data-test="calculator-button"
          onClick={() => onSubmit(this.state.amount, this.state.date)}
          className={styles.calculate}
          fluid
          basic
          color="green"
        >
          Calculate
        </Button>
      </Segment>
    );
  }
}

export default Calculator;
