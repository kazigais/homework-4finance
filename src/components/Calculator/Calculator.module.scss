.form-segment {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 300px !important;
  .input {
    width: 100%;
    padding-bottom: 10px;
  }
}

.calculate {
  max-width: 100% !important;
}
