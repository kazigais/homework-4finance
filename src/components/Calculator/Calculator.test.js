import React from 'react';
import Calculator from './Calculator';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Calculator component', () => {

  let props;

  beforeEach(() => {
    props = {
      onSubmit: jest.fn(),
      title: '',
      errorMessage: '',
      date: new Date('Feb 28 2013 19:00:00 EST'),
    };
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<Calculator {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('calls onSubmit when button is clicked', () => {
    const wrapper = shallow(<Calculator {...props} />);
    wrapper.find('[data-test="calculator-button"]').simulate('click');
    expect(props.onSubmit).toBeCalled();
  });
});
