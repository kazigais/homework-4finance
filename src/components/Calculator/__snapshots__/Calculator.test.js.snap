// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component matches snapshot 1`] = `
<Segment
  className="form-segment"
  raised={true}
>
  <Header
    as="h1"
  />
  <Input
    className="input"
    labelPosition="right"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
  >
    <Label
      basic={true}
    >
      <FaDollarSign />
    </Label>
    <input />
    <Label>
      .00
    </Label>
  </Input>
  <CalendarInput
    onChange={[Function]}
    value={2013-03-01T00:00:00.000Z}
  />
  <Button
    as="button"
    basic={true}
    className="calculate"
    color="green"
    data-test="calculator-button"
    fluid={true}
    onClick={[Function]}
  >
    Calculate
  </Button>
</Segment>
`;
